core/src/Kokkos_CopyViews.hpp

```diff
@@ -3765,11 +3785,24 @@ typename Kokkos::View<T, P...>::HostMirror create_mirror_view(
   return Impl::create_mirror_view(v, view_alloc(wi));
 }
 
+// FIXME_C++17 Improve SFINAE here.
 template <class Space, class T, class... P,
-          typename Enable = std::enable_if_t<Kokkos::is_space<Space>::value>>
+          class Enable = std::enable_if_t<Kokkos::is_space<Space>::value>>
 typename Impl::MirrorViewType<Space, T, P...>::view_type create_mirror_view(
-    Space const& space, Kokkos::View<T, P...> const& v) {
-  return Impl::create_mirror_view(space, v, Impl::ViewCtorProp<>{});
+    const Space&, const Kokkos::View<T, P...>& src,
+    std::enable_if_t<Impl::MirrorViewType<Space, T, P...>::is_same_memspace>* =
+        nullptr) {
```

Comment:
Doing the SFINAE in two separate places in the same signature (was used as a template parameter and now also as an additional function function argument) is hurting readability.